app/uk/gov/hmrc/tai/model/domain/income/TaxCodeIncome.scala

```diff
@@ -103,8 +103,40 @@ case class TaxCodeIncome(
   inYearAdjustmentIntoCYPlusOne: BigDecimal,
   iabdUpdateSource: Option[IabdUpdateSource] = None,
   updateNotificationDate: Option[LocalDate] = None,
-  updateActionDate: Option[LocalDate] = None)
+  updateActionDate: Option[LocalDate] = None) {
+
+  lazy val taxCodeWithEmergencySuffix: String = basisOperation match {
+    case Week1Month1BasisOperation => taxCode + TaiConstants.EmergencyTaxCode
+    case _                         => taxCode
+  }
+}
 
 object TaxCodeIncome {
+
+  implicit val writes = new Writes[TaxCodeIncome] {
+    override def writes(o: TaxCodeIncome): JsValue =
+      JsObject(
+        List(
+          "componentType"                 -> Json.toJson(o.componentType),
+          "employmentId"                  -> Json.toJson(o.employmentId),
+          "amount"                        -> Json.toJson(o.amount),
+          "description"                   -> Json.toJson(o.description),
+          "taxCode"                       -> Json.toJson(o.taxCodeWithEmergencySuffix),
+          "name"                          -> Json.toJson(o.name),
+          "basisOperation"                -> Json.toJson(o.basisOperation),
+          "status"                        -> Json.toJson(o.status),
+          "inYearAdjustmentIntoCY"        -> Json.toJson(o.inYearAdjustmentIntoCY),
+          "totalInYearAdjustment"         -> Json.toJson(o.totalInYearAdjustment),
+          "inYearAdjustmentIntoCYPlusOne" -> Json.toJson(o.inYearAdjustmentIntoCYPlusOne),
+          "iabdUpdateSource"              -> Json.toJson(o.iabdUpdateSource),
+          "updateNotificationDate"        -> Json.toJson(o.updateNotificationDate),
+          "updateActionDate"              -> Json.toJson(o.updateActionDate)
+        ).filter {
+          case (_, JsNull) => false
+          case _           => true
+        }
+      )
+  }
+
   implicit val format: Format[TaxCodeIncome] = Json.format[TaxCodeIncome]
```

Comment:
I'm not sure which writes would be chosen between the implicit writes or the implicit format. It would be better just remove any ambiguity and just use a pure reads
```suggestion
  implicit val reads: Reads[TaxCodeIncome] = Json.reads[TaxCodeIncome]
```